File: bot-api/java/src/main/java/dev/robocode/tankroyale/botapi/events/ScannedBotEvent.java
package dev.robocode.tankroyale.botapi.events;

/**
 * Event occurring when a bot has scanned another bot.
 */
@SuppressWarnings("unused")
public final class ScannedBotEvent extends BotEvent {

    // id of the bot did the scanning.
    private final int scannedByBotId;

    // id of the bot that was scanned.
    private final int scannedBotId;

    // Energy level of the scanned bot.
    private final double energy;

    // X coordinate of the scanned bot.
    private final double x;

    // Y coordinate of the scanned bot.
    private final double y;

    // Direction in degrees of the scanned bot.
    private final double direction;

    // Speed measured in units per turn of the scanned bot.
    private final double speed;

    /**
     * Initializes a new instance of the ScannedBotEvent class.
     *
     * @param turnNumber     is the turn number when the bot was scanned.
     * @param scannedByBotId is the id of the bot did the scanning.
     * @param scannedBotId   is the id of the bot that was scanned.
     * @param energy         is the energy level of the scanned bot.
     * @param x              is the X coordinate of the scanned bot.
     * @param y              is the Y coordinate of the scanned bot.
     * @param direction      is the direction in degrees of the scanned bot.
     * @param speed          is the speed measured in units per turn of the scanned bot.
     */
    public ScannedBotEvent(
            int turnNumber,
            int scannedByBotId,
            int scannedBotId,
            double energy,
            double x,
            double y,
            double direction,
            double speed) {
        super(turnNumber);
        this.scannedByBotId = scannedByBotId;
        this.scannedBotId = scannedBotId;
        this.energy = energy;
        this.x = x;
        this.y = y;
        this.direction = direction;
        this.speed = speed;
    }

    /**
     * Returns the id of the bot did the scanning.
     *
     * @return The id of the bot did the scanning.
     */
    public int getScannedByBotId() {
        return scannedByBotId;
    }

    /**
     * Returns the id of the bot that was scanned.
     *
     * @return The id of the bot that was scanned.
     */
    public int getScannedBotId() {
        return scannedBotId;
    }

    /**
     * Returns the energy level of the scanned bot.
     *
     * @return The energy level of the scanned bot.
     */
    public double getEnergy() {
        return energy;
    }

    /**
     * Returns the X coordinate of the scanned bot.
     *
     * @return The X coordinate of the scanned bot.
     */
    public double getX() {
        return x;
    }

    /**
     * Returns the Y coordinate of the scanned bot.
     *
     * @return The Y coordinate of the scanned bot.
     */
    public double getY() {
        return y;
    }

    /**
     * Returns the direction in degrees of the scanned bot.
     *
     * @return The direction in degrees of the scanned bot.
     */
    public double getDirection() {
        return direction;
    }

    /**
     * Returns the Speed measured in units per turn of the scanned bot.
     *
     * @return The Speed measured in units per turn of the scanned bot.
     */
    public double getSpeed() {
        return speed;
    }
}

File: bot_api_py/events/scanned_bot_event.py
from bot_api_py.events.bot_event import BotEvent


class ScannedBotEvent(BotEvent):
    """Event occurring when a bot has scanned another bot."""

    def __init__(self, turn_number: int, scanned_by_bot_id: int, scanned_bot_id: int, energy: float, x: float, y: float, direction: float, speed: float):
        """Initializes a new instance of the ScannedBotEvent class.

        Args:
            turn_number: The turn number when the bot was scanned.
            scanned_by_bot_id: The id of the bot did the scanning.
            scanned_bot_id: The id of the bot that was scanned.
            energy: The energy level of the scanned bot.
            x: The X coordinate of the scanned bot.
            y: The Y coordinate of the scanned bot.
            direction: The direction in degrees of the scanned bot.
            speed: The speed measured in units per turn of the scanned bot.
        """
        super().__init__(turn_number)
        self.scanned_by_bot_id = scanned_by_bot_id
        self.scanned_bot_id = scanned_bot_id
        self.energy = energy
        self.x = x
        self.y = y
        self.direction = direction
        self.speed = speed

    def get_scanned_by_bot_id(self) -> int:
        """Returns the id of the bot did the scanning.

        Returns:
            The id of the bot did the scanning.
        """
        return self.scanned_by_bot_id

    def get_scanned_bot_id(self) -> int:
        """Returns the id of the bot that was scanned.

        Returns:
            The id of the bot that was scanned.
        """
        return self.scanned_bot_id

    def get_energy(self) -> float:
        """Returns the energy level of the scanned bot.

        Returns:
            The energy level of the scanned bot.
        """
        return self.energy

    def get_x(self) -> float:
        """Returns the X coordinate of the scanned bot.

        Returns:
            The X coordinate of the scanned bot.
        """
        return self.x

    def get_y(self) -> float:
        """Returns the Y coordinate of the scanned bot.

        Returns:
            The Y coordinate of the scanned bot.
        """
        return self.y

    def get_direction(self) -> float:
        """Returns the direction in degrees of the scanned bot.

        Returns:
            The direction in degrees of the scanned bot.
        """
        return self.direction

    def get_speed(self) -> float:
        """Returns the Speed measured in units per turn of the scanned bot.

        Returns:
            The Speed measured in units per turn of the scanned bot.
        """
        return self.speed
